File: app/src/main/java/com/example/problem2/network/ApiManager.java
package com.example.problem2.network;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

public class ApiManager {
    static Gson gson = new GsonBuilder()
            .setLenient()
            .create();
    public static  Retrofit retrofit = new Retrofit.Builder()
            .baseUrl("https://jsonplaceholder.typicode.com/")
            .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
            .addConverterFactory(GsonConverterFactory.create(gson))
            .build();
    public static PostService postService = ApiManager.retrofit.create(PostService.class);
}

File: app/src/main/java/com/example/problem2/network/ApiManager.kt
package com.example.problem2.network


import com.google.gson.GsonBuilder
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

object ApiManager {
    private var gson = GsonBuilder()
        .setLenient()
        .create()

    private var retrofit = Retrofit.Builder()
        .baseUrl("https://jsonplaceholder.typicode.com/")
        .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
        .addConverterFactory(GsonConverterFactory.create(gson))
        .build()

    var postService: PostService = retrofit.create(PostService::class.java)
}